obiba_mica/import_zip.py

```diff
@@ -3,65 +3,145 @@
-      return self.__make_request().content_upload(path).resource('/draft/studies/_import?%s' % query).send()
-
-  @classmethod
-  def __printResponse(cls, response):
-    res = response.content
-    # output to stdout
-    if len(res) > 0:
-        print(res)
-
-  @classmethod
-  def add_arguments(cls, parser):
-      """
-      Add REST command specific options
-
-      :param parser - commandline args parser
-      """
-      parser.add_argument('path', help='Path to the zip file or directory that contains zip files to be imported')
-      parser.add_argument('--publish', '-pub', action='store_true', help='Publish imported study')
-
-  @classmethod
-  def do_command(cls, args):
-      """
-      Execute Import Zip command
-
-      :param args - commandline args
-      """
-
-      service = FileImportService(MicaClient.build(MicaClient.LoginInfo.parse(args)), args.verbose)
-      if args.path.endswith('.zip'):
-          cls.__printResponse(service.import_zip(args.path, args.publish))
-      else:
-          for export_file in os.listdir(args.path):
-              if export_file.endswith('.zip'):
-                  cls.__printResponse(service.import_zip(os.path.join(args.path, export_file), args.publish))
+    def __init__(self, client: MicaClient, verbose: bool = False):
+        self.client = client
+        self.verbose = verbose
+        self.needsLegacySupport = FileImportService.needsLegacySupport(client, verbose)
+
+    def __make_request(self):
+        request = self.client.new_request()
+        request.method("POST")
+        request.fail_on_error()
+        request.accept_json()
+        if self.verbose:
+            request.verbose()
+        return request
+
+    def __upgradeStudy(self, study):
+        study.pop("obiba.mica.CollectionStudyDto.type", None)
+        if "obiba.mica.EntityStateDto.studySummaryState" in study:
+            study["state"] = study["obiba.mica.EntityStateDto.studySummaryState"]
+        study.pop("obiba.mica.EntityStateDto.studySummaryState", None)
+
+    def __upgradeDataset(self, dataset):
+        if "obiba.mica.StudyDatasetDto.type" in dataset:
+            dataset["collected"] = dataset["obiba.mica.StudyDatasetDto.type"]
+            dataset.pop("obiba.mica.StudyDatasetDto.type", None)
+        else:
+            dataset["protocol"] = dataset["obiba.mica.HarmonizedDatasetDto.type"]
+            dataset.pop("obiba.mica.HarmonizedDatasetDto.type", None)
+
+    @staticmethod
+    def needsLegacySupport(client, verbose=False):
+        restService = RestService(client, verbose)
+        response = restService.send_request("/auth/session/_current", restService.make_request("GET"))
+        # Only the Mica server supporting Java 21 has version information in the response
+        versionInfo =  response.version_info
+
+        if versionInfo is not None:
+            return int(versionInfo["major"]) + int(versionInfo["minor"]) < 10
```

Comment:
What does that sum means?

Reply:
Now that it passes the checks, can you explain this one?

Reply:
I have changed  the code to be more readable, the old code meant to check for `vesions < 5.5.x`.

Reply:
But 6 + 0 is lower than 10

Reply:
I have no longer this code, here is the replacement:

```
return int(versionInfo["major"]) <= 5 and int(versionInfo["minor"]) < 5
```